### PR TITLE
made links to queries

### PR DESCRIPTION
**Pull request by holly-th**
I didn't test my be to have both a sortby and orderby so they don't work together unfortunately but do work individually. I only had the ascending option as it is automatically descenting but i may add another button for descending that takes away the orderby
Added a descending option

### CODE REVIEW

File: src/Articles.jsx [hunk 1]
```diff
@@ -6,20 +6,30 @@ function Articles() {
...
     <section>
+      orderby:
+      <Link to={"?orderby=asc"}>Ascending</Link>
+      <Link to={"/"}>Descending</Link>
+      Sort by:
+      <Link to={"/articles?sortby=author"}>Author</Link>
+      <Link to={"/articles?sortby=topic"}>Topic</Link>
+      <Link to={"/articles?sortby=created_at"}>Date</Link>
+      <Link to={"/articles?sortby=votes"}>Votes</Link>
+      <Link to={"/articles?sortby=article_id"}>Article ID</Link>
```
Comment (dav3rid) [inline src/Articles.jsx]: These links work well - they are however, as you've mentioned, restricting your FE from accounting for multiple queries in the URL.
Take a look at the notes: [here](https://notes.northcoders.com/courses/js-front-end/react-router#updating-queries) for this

File: src/utils/api.js [hunk 2]
```diff
@@ -3,11 +3,13 @@ const newsApi = axios.create({
...
 });
 
-export function getArticles(topic) {
+export function getArticles(topic, sortby, orderby) {
   return newsApi
     .get("/api/articles", {
       params: {
         topic: topic,
+        sortby: sortby,
+        orderby: orderby,
```
Comment (dav3rid) [inline src/utils/api.js:12]: Nice use of axios params to handle multiple queries

**Review by dav3rid — APPROVED** [review]
You mentioned that you can only provide one query at a time - you'll probably find that your BE can accept multiple queries - but if it's not tested no need to put it in place yet.

You may also find that the way you have your useEffect set up along side extracting the 3 queries you care about from the URL, you can handle multiple queries when typing directly in the URL. Eg. `?topic=football&sortby=title&orderby=asc` works fine - some comments left regarding your links to get this to also work without typing this manually

File: src/Articles.jsx [hunk 3]
```diff
@@ -4,22 +4,86 @@ import { Link, useSearchParams } from "react-router-dom";
...
       setIsLoading(false);
     });
-  }, [topic]);
+  }, [topic, sortby, orderby]);
 
   return isLoading ? (
     <p>Loading Articles...</p>
   ) : (
     <section>
+      orderby:
```
Comment (JVal-NC) [inline src/Articles.jsx]: This seems to be working nice but would say the UI is a tad ambiguous. Maybe one to do if styling later on but visually it isnt clear which sortby/orderby buttons have been clicked. Would a different html component be better suited than a button?

**Review by JVal-NC — COMMENTED** [review]
Hi Holly, look good. Some comments to consider but go ahead and merge

**Pull request merged**